docker-compose.yml

```diff
@@ -9,8 +9,28 @@ x-services:
       - postgres
       - minio
     environment:
-      DATABASE_URI: postgresql+psycopg2://$DB_USER:$DB_PASSWORD@$DB_HOST:$CONN_DB_PORT/$DB_NAME
-      API_PORT: $API_PORT
+      - DATABASE_URI=postgresql+psycopg2://$DB_USER:$DB_PASSWORD@$DB_HOST:$CONN_DB_PORT/$DB_NAME
+      - API_PORT=$API_PORT
```

Comment:
For me, these environment variables aren't inherited by the extended services.
I can't find an exact reference in a manual for this quote "merge is only supported by the YAML specifications for mappings and not for sequences" (https://stackoverflow.com/questions/24090177/how-to-merge-yaml-arrays)

But I can tell you that `API_PORT` and `DATABASE_URI` were empty in the api service before these changes for me.

Reply:
Maybe we could try extending base services as explained here: https://docs.docker.com/compose/extends/#understand-the-extends-configuration

Reply:
it's for compose v2... :(

Reply:
Nah, it's no longer available:
> The extends keyword is supported in earlier Compose file formats up to Compose file version 2.1 (see extends in v1 and extends in v2), but is not supported in Compose version 3.x.